key.ui/src/main/java/de/uka/ilkd/key/gui/docking/DockingLayout.java

```diff
@@ -140,102 +126,17 @@ private void setLayout(String layout) {
         DockingHelper.restoreMissingPanels(window);
     }
 
-    @Nonnull
-    @Override
-    public List<JMenuItem> getMainMenuItems(@Nonnull MainWindow mainWindow) {
-        List<JMenuItem> items = new ArrayList<>();
-
-        final class ActivateLayoutAction extends MainWindowAction {
-            private final String layout;
-
-            private ActivateLayoutAction(MainWindow mainWindow, String layout) {
-                super(mainWindow);
-                this.layout = layout;
-                setName(layout);
-                setMenuPath("View.Layout");
-            }
-
-            @Override
-            public void actionPerformed(ActionEvent e) {
-                setLayout(layout);
-                layouts.getElements().asIterator().forEachRemaining(
-                    button -> button.getModel().setSelected(button.getText().contains(layout)));
-            }
-        }
-
-        for (String s : LAYOUT_NAMES) {
-            JRadioButtonMenuItem button = new JRadioButtonMenuItem(s);
-            if (s.equals("Default")) {
-                button.getModel().setSelected(true);
-            } else {
-                button.getModel().setSelected(false);
-            }
-            layouts.add(button);
-            button.setAction(new ActivateLayoutAction(mainWindow, s));
-            items.add(button);
-        }
-        return items;
-    }
-
     @Override
     public List<Action> getMainMenuActions(MainWindow mainWindow) {
         List<Action> actions = new ArrayList<>();
-
-        final class SaveAction extends MainWindowAction {
-            private static final long serialVersionUID = -2688272657370615595L;
-
-            private SaveAction(MainWindow mainWindow) {
-                super(mainWindow);
-                setName("Save Layout");
-                setMenuPath("View.Layout");
-            }
-
-            @Override
-            public void actionPerformed(ActionEvent e) {
-                String layout = null;
-                var iter = layouts.getElements().asIterator();
-                while (iter.hasNext()) {
-                    var b = (JRadioButtonMenuItem) iter.next();
-                    if (b.getModel().isSelected()) {
-                        layout = b.getText();
-                        System.out.println("saving in " + layout);
-                        break;
-                    }
-                }
-                mainWindow.getDockControl().save(layout);
-            }
+        int keypos = 0;
+        for (String layout : LAYOUT_NAMES) {
+            Integer key = keypos < LAYOUT_KEYS.length ? LAYOUT_KEYS[keypos] : null;
+            actions.add(new LoadLayoutAction(mainWindow, layout, key));
+            actions.add(new SaveLayoutAction(mainWindow, layout, key));
+            keypos++;
```

Comment:
```suggestion
        for (String layout : LAYOUT_NAMES) {
            Integer key = keypos < LAYOUT_KEYS.length ? LAYOUT_KEYS[keypos] : null;
            actions.add(new LoadLayoutAction(mainWindow, layout, key));
            keypos++;
        }
        for (String layout : LAYOUT_NAMES) {
            Integer key = keypos < LAYOUT_KEYS.length ? LAYOUT_KEYS[keypos] : null;
            actions.add(new SaveLayoutAction(mainWindow, layout, key));
            keypos++;
```

Grouping the menu entries by slot makes it hard to read in my opinion:
![image](https://github.com/KeYProject/key/assets/94171076/23b52c57-700e-481e-a259-e093b39e7e64)

My suggestion groups them by action:
![image](https://github.com/KeYProject/key/assets/94171076/e60e1df1-9102-4e33-8a5d-9214bbb89ad4)